components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoFinalityFlowV1.kt

```diff
@@ -100,14 +116,62 @@ class UtxoFinalityFlowV1(
     private fun sendTransactionAndBackchainToCounterparties(transferAdditionalSignatures: Boolean) {
         flowMessaging.sendAll(FinalityPayload(initialTransaction, transferAdditionalSignatures), sessions.toSet())
 
-        sessions.forEach {
-            if (initialTransaction.dependencies.isNotEmpty()) {
-                flowEngine.subFlow(TransactionBackchainSenderFlow(initialTransaction.id, it))
-            } else {
-                log.trace {
-                    "Transaction with id ${initialTransaction.id} has no dependencies so backchain resolution will not be performed."
+        val notaryInfo = requireNotNull(notaryLookup.lookup(initialTransaction.notaryName)) {
+            "Notary ${initialTransaction.notaryName} does not exist in the network"
+        }
+        if (notaryInfo.isBackchainRequired) {
+            sessions.forEach {
+                if (initialTransaction.dependencies.isNotEmpty()) {
+                    flowEngine.subFlow(TransactionBackchainSenderFlow(initialTransaction.id, it))
+                } else {
+                    log.trace {
+                        "Transaction with id ${initialTransaction.id} has no dependencies so backchain resolution will not be performed."
+                    }
                 }
             }
+        } else {
+            val filteredTransactionsAndSignatures = initialTransaction
+                .let { it.inputStateRefs + it.referenceStateRefs }
+                .groupBy { stateRef -> stateRef.transactionId }
+                .mapValues { (_, stateRefs) -> stateRefs.map { stateRef -> stateRef.index } }
+                .map { (transactionId, indexes) ->
+                    val dependency = requireNotNull(utxoLedgerService.findSignedTransaction(transactionId)) {
+                        "Dependent transaction $transactionId does not exist"
+                    }
+                    val newTxNotaryKey = initialTransaction.notaryKey
+                    val dependencyNotaryKey = dependency.notaryKey
+                    require(initialTransaction.notaryName == dependency.notaryName) {
+                        "Notary name of filtered transaction \"${dependency.notaryName}\" doesn't match with " +
+                            "any names of initial transaction \"${initialTransaction.notaryName}\""
```

Comment:
That error message does not make sense anymore if you compare the notary service name rather than the representatives' names.
```suggestion
                        "Notary service \"${dependency.notaryName}\"  of input transaction ${filteredTransaction.id} does not match " +
                            "notary service of current transaction \"${initialTransaction.notaryName}\""
```